### PR TITLE
AO-11811: Trace decision sync-up

### PR DESCRIPTION
**Pull request by jiwen624**
Negotiation is required to merge the remote settings with the local sampling config.

See https://swicloud.atlassian.net/browse/AO-11811 for more details.

### CODE REVIEW

File: v1/ao/internal/reporter/oboe.go [hunk 1]
```diff
@@ -292,27 +266,63 @@ func parseInt32(args map[string][]byte, key string, fb int32) int32 {
 	return ret
 }
 
+// mergeLocalSetting follow the predefined precedence to decide which one to
+// pick from: either the local configs or the remote ones, or the combination.
+//
+// Note: This function modifies the argument in place.
+func mergeLocalSetting(remote *oboeSettings) *oboeSettings {
```
Comment (jiwen624) [inline v1/ao/internal/reporter/oboe.go]: The precedence is supposed to be:
if both remote (with override flag) and local, lowest setting wins -- lowest is pretty obvious re: sample rate, and we're just saying "and-ed together" for now for the tracing flags. OR
if both remote (without override flag) and local, local setting wins. OR
remote settings

**Comment by jiwen624** [comment]
> I think we changed tracing mode ALWAYS to ENABLED and NEVER to DISABLED. Should this be reflected in the code change?

Thanks for catching that! Done.

**Comment by dschnabel** [comment]
Looks good :+1:

**Comment by jiwen624** [comment]
Fixed a problem found in integration testing.

**Pull request merged**